### PR TITLE
fix:  correct validation for trust relationship request

### PR DESCRIPTION
**Pull request by pranavkparti**
## Description
Fixed validation while creating new trust relationship request. 
- Subwallet cannot send request to its own parent 

**What kind of change(s) does this PR introduce?**

- [ ] Enhancement
- [x] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Breaking change

**Does this PR introduce a breaking change?**
No

## Other useful information

None

### CODE REVIEW

**Comment by Kpoke** [comment]
Can you fix the tests? @pranavkparti

**Comment by pranavkparti** [comment]
> Can you fix the tests? @pranavkparti

It seems to be some issue with the workflow itself. @samwel141 's PR also fails with the same errors.

File: server/models/Trust.js [hunk 1]
```diff
@@ -141,23 +141,36 @@ class Trust {
...
 
+    // check if actor can control the target
+    const actorHasControlOverTarget = await walletModel.hasControlOver(
+      actorWallet.id,
+      targetWallet.id,
+    );
+
     // originating wallet doesn't need to send requests to a sub wallet it manages
-    if (hasControlOverTarget) {
+    if (actorHasControlOverTarget) {
```
Comment (Kpoke) [inline server/models/Trust.js:160]: Why was this replaced with actorHasControlOverTarget?

Comment (pranavkparti) [reply server/models/Trust.js:160]: O = Originator, A = Actor, T = Target
 
By the time the code reaches here, the check is made that O != A && O != T && O has control over both A and T. Meaning that A and T are subwallets of O, and an error displays if it is true.

Next a check is made that A has control over T. This is to deal with the case if A = O, and has control over T. Meaning T (requestee)  is already managed by A (requester). 

Next, the last check is made that T = O. Meaning that A is a subwallet of O, and  target is the originating wallet.  Meaning T (requestee) already manages A (requester).

@Kpoke The checks are mainly for displaying specific and appropriate error messages.

Comment (Kpoke) [reply server/models/Trust.js:160]: Okay noted. That's fine. I just wanted to confirm

Comment (Kpoke) [reply server/models/Trust.js:160]: Can you pull the latest changes I made to your PR? It'll enable the PR github action run

Comment (pranavkparti) [reply server/models/Trust.js:171]: @Kpoke Done, thank you

**Pull request merged**